python/lsst/ip/isr/shutterMotion.py

```diff
@@ -0,0 +1,576 @@
+# This file is part of ip_isr.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+Shutter motion profile storage class
+"""
+
+__all__ = ["ShutterMotionProfile"]
+
+from astropy.table import Table
+from scipy.optimize import newton
+from numpy import nan
+
+from lsst.ip.isr import IsrCalib
+
+
+class ShutterMotionProfile(IsrCalib):
+    """Shutter motion profile measurements.
+
+    Parameters
+    ----------
+    log : `logging.Logger`, optional
+        Log to write messages to. If `None` a default logger will be used.
+    **kwargs :
+        Additional parameters.
+    """
+
+    _OBSTYPE = 'shutterMotionProfile'
+    _SCHEMA = 'ShutterMotionProfile'
+    _VERSION = 1.0
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        # Quantities that come from `encodeSamples`
+        self.time_tai = []
+        self.time_mjd = []
+        self.position = []
+        self.hall_time_tai = []
+        self.hall_time_mjd = []
+        self.hall_position = []
+        self.hall_sensorId = []
+        self.hall_isOn = []
+        self.fit_name = []
+        self.fit_start_time = []
+        self.fit_pivot1 = []
+        self.fit_pivot2 = []
+        self.fit_jerk0 = []
+        self.fit_jerk1 = []
+        self.fit_jerk2 = []
+
+        self.requiredAttributes.update(['time_tai', 'time_mjd', 'position',
```

Comment:
There is a mix of single-quote and double-quote in this file.  I would request that you settle on one (and sadly, double-quote has won out in general).